sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableChange.java

```diff
@@ -787,4 +806,81 @@ public int hashCode() {
       return Arrays.hashCode(clusteringColumns);
     }
   }
+
+  /** A TableChange to alter table and add a constraint. */
+  final class AddConstraint implements TableChange {
+    private final Constraint constraint;
+    private final boolean validate;
+
+    private AddConstraint(Constraint constraint, boolean validate) {
+      this.constraint = constraint;
+      this.validate = validate;
+    }
+
+    public Constraint getConstraint() {
```

Comment:
Minor: I believe we rarely use `get` prefix in this file/set of APIs. I'd call it `constraint()`.